app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -434,13 +453,25 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
 
         reverseGeocodeData?.clear()
 
-        val simpleFeature = SimpleFeature.fromFeature(features.get(0))
-        val lngLat = LngLat(simpleFeature.lng(), simpleFeature.lat())
+        var lngLat: LngLat?
+
+        if (poiTapPoint == null) {
+            val simpleFeature = SimpleFeature.fromFeature(features.get(0))
+            lngLat = LngLat(simpleFeature.lng(), simpleFeature.lat())
+        } else {
+            val point = poiTapPoint
+            lngLat = mapController?.coordinatesAtScreenPosition(point!![0].toDouble(), point!![1].toDouble())
```

Comment:
Using `!!` is also kind of scary since its one of the few ways to get a null pointer exception in Kotlin. Something like this would be safer (but would also require updating line 467 below since `lngLat` now may not have been initialized).

```kotlin
            val lng = poiTapPoint?.get(0)?.toDouble()
            val lat = poiTapPoint?.get(1)?.toDouble()
            if (lng != null && lat != null) {
                lngLat = mapController?.coordinatesAtScreenPosition(lng, lat)
            }
```